### ISSUE TITLE
'Material Icons' is not a system font

### ISSUE THREAD

**Issue by alexborton** [react-native-training/react-native-elements#1005]
I can't seem to get Material Icons font to render at all. I have read a lot of threads on the matter and nothing seems to work.

I am running on Expo and have the following setup in my `App.js`

```
import React from 'react'
import { Font, AppLoading } from 'expo'
import { MaterialIcons } from '@expo/vector-icons'
import HybridApp from './src/App'

export default class NativeApp extends React.Component {
  constructor() {
    super()
    this.state = {
      fontsAreLoaded: false
    }
  }
  async componentWillMount() {
    await Font.loadAsync(MaterialIcons.font)
    this.setState({ fontsAreLoaded: true })
  }
  render() {
    const { fontsAreLoaded } = this.state
    return !fontsAreLoaded ? <AppLoading /> : <HybridApp />
  }
}
```

This is as per the Expo docs and suggested in some other issues on React native elements. Yet still i get the same RSD;

![image](https://user-images.githubusercontent.com/1439507/37496754-4d3d9b8e-28ac-11e8-828b-4e6203dd4565.png)

I have not ejected my app and don't think i should have to with the expo/vector-icons as they support Material Icons?

**Comment by xavier-villelegier** [comment]
@alexborton Did you try with `@expo/vector-icons` ?
```
npm install --save "@expo/vector-icons"
```

and then load your font from

```
import { MaterialIcons } from '@expo/vector-icons'; 
```

**Comment by alexborton** [comment]
@xavier-villelegier Yes - sorry i did try that (I have re-formatted my original question code)

```
import React from 'react'
import { Font, AppLoading } from 'expo'
import { MaterialIcons } from '@expo/vector-icons'
import HybridApp from './src/App'

export default class NativeApp extends React.Component {
  constructor() {
    super()
    this.state = {
      fontsAreLoaded: false
    }
  }
  async componentWillMount() {
    await Font.loadAsync(MaterialIcons.font)
    this.setState({ fontsAreLoaded: true })
  }
  render() {
    const { fontsAreLoaded } = this.state
    return !fontsAreLoaded ? <AppLoading /> : <HybridApp />
  }
}
```

I read about @expo/vector-icons exporting Material icons as `material` so also tried;

`await Font.loadAsync({'MaterialIcons': require('@expo/vector-icons/fonts/MaterialIcons.ttf')})
`

**Comment by alexborton** [comment]
After hours wracking my brain on this, the answer was there in front of me the whole time. 

Presumably, React Native Elements refers to Material icons as `Material Icons`, *not* `MaterialIcons`.

This means that the default import from `@expo/vector-icons` does not work as their reference to Material icons is different.

The solution is to manually select Material icons from expo, replacing this line;

`await Font.loadAsync(MaterialIcons.font)` with

`await Font.loadAsync({
  'Material Icons': require('@expo/vector-icons/fonts/MaterialIcons.ttf')
})`

I hope this saves someone some time in the future.

**Comment by moshfeu** [comment]
@alexborton Thanks!

**Comment by kriskros09** [comment]
@alexborton Thanks !! you saved me a lot of time !

**Comment by feargswalsh92** [comment]
Started working on integrating this into the library. Recreated the bug in a[ test project ](https://github.com/feargswalsh92/1005.git )and am made [a fork](https://github.com/feargswalsh92/react-native-elements/tree/1005)  of the library that I'm using to test my changes are fixing this. It seems to me that in order to fix this I would need to add the logic that @alexborton  suggested to  [getIconType.js](https://github.com/react-native-training/react-native-elements/blob/master/src/helpers/getIconType.js), anybody know if this is correct?

**Comment by cbartram** [comment]
@alexborton You saved me so much time I spent hours on this and this was the solution I was looking for thank you so much!

**Comment by robreuss** [comment]
@alexborton Your fix was working great for me on version 0.19.1 but unfortunately the message is back again on 1.0.0 beta 5.  Any work arounds?  Thank you!  (I'm wanting to use the Overlay component which is why I'm looking to use the beta.)

**Comment by robreuss** [comment]
@alexborton I think I've answered my own question.  The string needs to be changed to `Material Design Icons`, as in:

`await Expo.Font.loadAsync({ "Material Design Icons": require("@expo/vector-icons/fonts/MaterialIcons.ttf") });`

**Comment by iRoachie** [comment]
@feargswalsh92 Did you manage to get anything worked out?

**Comment by kalifa17** [comment]
`yarn add react-native-vector-icons`  solved the issue for me

**Comment by maxtuzz** [comment]
I'm still getting this issue and it's quite strange. 

In my app.js:

```js
loadResourcesAsync = async () => {
    return Promise.all([
      Asset.loadAsync([
        require('../assets/images/robot-dev.png'),
        require('../assets/images/robot-prod.png'),
      ]),
      Font.loadAsync({
        'Material Icons': require('@expo/vector-icons/fonts/MaterialIcons.ttf'),
        // This is the font that we are using for our tab bar
        ...Icon.Ionicons.font,
        // We include SpaceMono because we use it in PickupScreen.js. Feel free
        // to remove this if you are not using it in your app
        'space-mono': require('../assets/fonts/SpaceMono-Regular.ttf'),
      }),
    ]);
  };
```

And in my RandomScreen.js:

```js
<View>
        {
          list.map((item, i) => (
            <ListItem
              key={i}
              title={item.title}
              leftIcon={{ name: item.icon }}
            />
          ))
        }
      </View>
```

And I still get this issue. Using react-navigation to render screens based on tabs.

**Comment by iRoachie** [comment]
@maxtuzz In a new project, I didn't have to use Font.loadAsync. The icons just worked out of the box.

```
"expo": "^32.0.0",
"react-native-elements": "^1.0.0"
```

**Comment by ramasubbaiya** [comment]
@iRoachie I was using expo ~30.0.0 then upgraded to 32.0.3 but still it did not work for me. 

I am using `<Avatar />` from _react-native-elements_

```
<Avatar
   size="xlarge" 
   rounded title="CR" 
   onPress={() => console.log('Works!')} 
   activeOpacity={0.7} 
   showEditButton
/>
```

`showEditButton` requires Material Icon Font

**Comment by iRoachie** [comment]
I just opened a project using  "expo": "^31.0.2" and it worked out the box

**Comment by kajchang** [comment]
For me, it was working without having the load icons but stop working today with no changes.

**Comment by gerwim** [comment]
So, my app required both `MaterialIcons` and `Material Icons` (note the space - which I didn't see for an hour)...

**Comment by vidurajith-darshana** [comment]
@alexborton thanks a lot, you saved my time!

**Comment by aacassandra** [comment]
@ramasubbaiya not material icon but MaterialCommunityIcons

**Comment by ravirajar** [comment]
Anyone still facing this issue? I'm not able to sort this issue still.

**Comment by nonameolsson** [comment]
Yeah, I have been struggling with this problem the last days. After much debugging I think I found the problem.

The solution was very simple. I have a `babel.config.js`, and had to add ` "babel-preset-expo"` to `presets`.

After that I removed `node_modules`, `.expo` and `yarn.lock`. After that I installed modules with `yarn` and then started the packager with `expo start -c`. 

`babel.config.js`

```js
module.exports = {
  presets: ["module:metro-react-native-babel-preset", "babel-preset-expo"],
  env: {
    production: {},
  },
  plugins: [
    [
      "@babel/plugin-proposal-decorators",
      {
        legacy: true,
      },
    ],
    ["@babel/plugin-proposal-optional-catch-binding"],
  ],
}

```

**Comment by Zac-Evans** [comment]
Looks like the font directory has been changed in @expo/vector-icons. Also, I'm using a functional component, so I used expo's useFonts hook. In my case, I was trying to use  [React Native Paper's Avatar](https://callstack.github.io/react-native-paper/avatar-icon.html) component and was getting a similar error. Here is the solution that worked for me:

```
import { useFonts } from 'expo-font';

let [fontsLoaded] = useFonts({
    'Material Design Icons': require('@expo/vector-icons/build/vendor/react-native-vector-icons/Fonts/MaterialCommunityIcons.ttf'),
  });
  
  if (!fontsLoaded) {
    return <Text>Loading...</Text>;
  }
  
  return (<YourComponent />)
```

**Comment by headwinds** [comment]
you can also [download MaterialCommunityIcons.ttf](https://github.com/oblador/react-native-vector-icons/blob/master/Fonts/MaterialCommunityIcons.ttf) and it to your local assets/fonts folder 

```
await Promise.all([
            Font.loadAsync({
              helvetica_bold: require('./assets/fonts/Helvetica-Bold.ttf'),
              helvetica_light: require('./assets/fonts/Helvetica-light.ttf'),
              helvetica: require('./assets/fonts/Helvetica.ttf'),
              'Material Design Icons': require('./assets/fonts/MaterialCommunityIcons.ttf'),
            }),
          ]);
```
